tests/ui/loop-match/const-continue-outside-loop-match.rs

```diff
@@ -0,0 +1,62 @@
+// Test that the correct error is emitted when `#[const_continue]` occurs outside of
+// a loop match. See also https://github.com/rust-lang/rust/issues/143165.
+#![allow(incomplete_features)]
+#![feature(loop_match)]
+#![crate_type = "lib"]
+
+fn main1() {
+    let mut state;
+    loop {
+        state = 'state: {
+            #[const_continue]
+            break break;
+            //~^ ERROR a `#[const_continue]` can only occur in a `#[loop_match]` loop
+            //~| ERROR unlabeled `break` inside of a labeled block
+            //~| ERROR unlabeled `break` inside of a labeled block
+        }
+    }
+}
+
+enum State2 {
+    X,
+    Z,
+}
+
+fn main2() {
```

Comment:
Does this example have to be that nested? Could it be simplified? If not, could you add a comment? It's hard to see what it is testing..